ast/src/parsed/display.rs

```diff
@@ -331,7 +331,7 @@ impl<T: Display> Display for Params<T> {
 
 impl<E: Display> Display for IndexAccess<Expression<E>> {
     fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-        if self.array.precedence().is_none() {
+        if self.array.precedence().is_none() || self.array.is_postfix() {
```

Comment:
I don't think we should do it like that. If you implement precedence for IndexAccess and FunctionCall, this should just work as the others: Parentheses if the precedence of the inner is higher than the current, otherwise no parentheses.

Reply:
Hm, ok, maybe this is the way to go, sorry :)